ex1/exceptions.py

```diff
@@ -46,3 +57,19 @@
 # the first by the second. This function should handle exceptions that might
 # occur print out what went wrong and return None if no results could be
 # computed.
+
+def safe_divide(a, b):
+    if not isinstance(a, numbers.Number):
```

Comment:
You don't have to check the instances you can just except the Error which is less work.